src/i18n.ts

```diff
@@ -410,6 +330,31 @@ export function invalidate(bundlePath?: string) {
 }
 
 /**
+ * Register an observer to be notified when the root locale changes.
+ *
+ * @param observer
+ * The observer whose `next` method will receive the locale string on updates, and whose `error` method will receive
+ * an Error object if the locale switch fails.
+ *
+ * @return
+ * A subscription object that can be used to unsubscribe from updates.
+ */
+export const observeLocale = (function () {
+	const localeSource = new Observable<string>((observer: SubscriptionObserver<string>) => {
+		localeProducer.on('change', (event) => {
+			observer.next(event.target as string);
+		});
+		localeProducer.on('error', (event: ErrorEvent) => {
+			observer.error(event.error);
+		});
+	});
```

Comment:
Do you want to return a clean up function? The function could be used to deregister the `change` and `error` event listeners when the subscription isn't needed anymore.